src/taus.cxx

```diff
@@ -7,18 +7,346 @@
+ * tau \f$p_T\f$ values
+ * @param pt name of the input column containing hadronic tau \f$p_T\f$ values
+ * @param eta name of the column containing hadronic tau eta values
+ * @param decay_mode name of the column containing hadronic tau decay modes
+ * @param gen_match name of the column with the matching information of the
+ * hadronic tau to generator-level particles (matches are: 1=prompt e, 2=prompt mu,
+ * 3=tau->e, 4=tau->mu, 5=had. tau, 0=unmatched)
+ * @param es_file path to the correction file for the energy scale correction
+ * @param correction_name name of the correction in `es_file`
+ * @param id_algorithm identification algorithm used for hadronic tau ID
+ * @param id_vs_jet_wp working point for the identification against jets; set to `""` if the corrections do not depend on this parameter
+ * @param id_vs_ele_wp working point for the identification against electrons; set to `""` if the corrections do not depend on this parameter
+ * @param variation_efake_dm0_barrel variation for electron faking a tau for decay mode 0 in the barrel region,
+ * options are "nom", "up", "down"
+ * @param variation_efake_dm1_barrel variation for electron faking a tau for decay mode 1 in the barrel region,
+ * options are "nom", "up", "down"
+ * @param variation_efake_dm0_endcap variation for electron faking a tau for decay mode 0 in the endcap regionefake_,
+ * options are "nom", "up", "down"
+ * @param variation_efake_dm1_endcap variation for electron faking a tau for decay mode 1 in the endcap region,
+ * options are "nom", "up", "down"
+ * @param variation_mufake variation for muon faking a tau, options are "nom", "up", "down"
+ * @param variation_gentau_dm0 variation for genuine tau for decay mode 0, options are "nom", "up",
+ * "down"
+ * @param variation_gentau_dm1 variation for genuine tau for decay mode 1, options are "nom", "up",
+ * "down"
+ * @param variation_gentau_dm10 variation for genuine tau for decay mode 10, options are "nom", "up",
+ * "down"
+ * @param variation_gentau_dm11 variation for genuine tau for decay mode 11, options are "nom", "up",
+ * "down"
+ * @return a dataframe containing the corrected transverse momenta
+ *
+ * @note This function is intended to be used for Run 3 analyses. In Run 3,
+ * the tau energy scale corrections also depend on the DeepTau working points
+ * for ID vs. electrons and vs. jets. An overloaded version of this function
+ * exists for this purpose.
+ */
+ROOT::RDF::RNode
+PtCorrectionMC(
+    ROOT::RDF::RNode df,
+    correctionManager::CorrectionManager &correction_manager,
+    const std::string &outputname,
+    const std::string &pt,
+    const std::string &eta,
+    const std::string &decay_mode,
+    const std::string &gen_match,
+    const std::string &es_file,
+    const std::string &correction_name,
+    const std::string &id_algorithm,
+    const std::string &id_vs_jet_wp,
+    const std::string &id_vs_ele_wp,
```

Comment:
I suggest to move these two parameters to the end of the inputs and directly setting the default values. By that run2 analyses don't need to specify these parameters with `""` by hand.